### PR TITLE
mixin-mixout: fix 2 corner cases

### PR DESCRIPTION
**Pull request by lyakh**
Fix a case of empty or inactive sources and `audio_stream_set_zero()` error handling

### CODE REVIEW

File: src/audio/mixin_mixout/mixin_mixout.c [hunk 1]
```diff
@@ -512,7 +512,9 @@ static int mixout_process(struct processing_module *mod,
 			continue;
 
 		pending_frames = md->pending_frames[source_index];
-		frames_to_produce = MIN(frames_to_produce, pending_frames);
+
+		if (source->state == COMP_STATE_ACTIVE || pending_frames)
+			frames_to_produce = MIN(frames_to_produce, pending_frames);
```
Comment (kv2019i) [inline src/audio/mixin_mixout/mixin_mixout.c:517]: I can understand the !COMP_STATE_ACTIVE case, but how come we can ignore zero pending frames case? Wouldn't this cause audio quality issue if the active source comes back with data on next 1ms tick, but we already mixed out zeros to the output?

**Review by kv2019i — COMMENTED** [review]
Other patches ok, but not sure about the behaviour change in the second patch. Please see question inline.

**Review by lgirdwood — APPROVED** [review]

File: src/audio/mixin_mixout/mixin_mixout.c (hunk 1 above)
Comment (lyakh) [reply src/audio/mixin_mixout/mixin_mixout.c:517]: @kv2019i you mean when that zero-pending source is active? Then we **will** fo the calculation and `frames_to_produce` will be 0

Comment (kv2019i) [reply src/audio/mixin_mixout/mixin_mixout.c:517]: Ack @lyakh , so logic is correct.

**Review by kv2019i — APPROVED** [review]
Thanks @lyakh , clear now.

**Pull request merged**